sound/soc/sof/pcm.c

```diff
@@ -241,7 +241,7 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	struct snd_sof_pcm *spcm = rtd->private;
 	struct sof_ipc_stream stream;
 	struct sof_ipc_reply reply;
-	int ret = 0;
+	int ret = 0, permit_d0i3;
```

Comment:
D0, D0i1, D3 are valid states
D0i3 isn't really defined anywhere. D0i2 in some cases but without audio.